src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py

```diff
@@ -8,27 +8,20 @@
 import requests
 from requests.utils import to_native_string
 
-from azure.cli.core.prompting import prompt, prompt_pass, NoTTYException, prompt_y_n
+from azure.cli.core.prompting import NoTTYException, prompt_y_n
 import azure.cli.core.azlogging as azlogging
 from azure.cli.core.util import CLIError
 
-from azure.mgmt.containerregistry.v2017_06_01_preview.models import SkuTier
-
-from ._utils import get_registry_by_name, managed_registry_validation
-from ._docker_utils import get_login_access_token
-from .credential import acr_credential_show
+from ._utils import managed_registry_validation
+from ._docker_utils import get_access_credentials
 
 
 logger = azlogging.get_az_logger(__name__)
-DELETE_NOT_SUPPORTED = 'Delete is not supported for registries in Basic SKU.'
-LIST_MANIFESTS_NOT_SUPPORTED = 'List manifests is not supported for registries in Basic SKU.'
+DELETE_NOT_SUPPORTED = 'Delete is only supported for managed registries.'
+LIST_MANIFESTS_NOT_SUPPORTED = 'List manifests is only supported for managed registries.'
 
 
-class NotFound(Exception):
-    pass
-
-
-class Unauthorized(Exception):
+class RetryableException(Exception):
```

Comment:
Why do you define a new type of exception while there isn't any mechanism to treat this exception differently (for example, `except RetryableException`.)?

Please avoid introducing new exception type unless there is specific logic build around it.